servers/rendering/renderer_rd/renderer_scene_render_rd.cpp

```diff
@@ -260,15 +260,30 @@ void RendererSceneRenderRD::_render_buffers_copy_screen_texture(const RenderData
 
 	RD::get_singleton()->draw_command_begin_label("Copy screen texture");
 
-	rb->allocate_blur_textures();
-
+	StringName texture_name;
 	bool can_use_storage = _render_buffers_can_be_storage();
 	Size2i size = rb->get_internal_size();
 
+	// When upscaling, the blur texture needs to be at the target size for post-processing to work. We prefer to use a
+	// dedicated backbuffer copy texture instead if the blur texture is not an option so shader effects work correctly.
+	Size2i target_size = rb->get_target_size();
+	bool using_fsr2 = (rb->get_scaling_3d_mode() == RS::VIEWPORT_SCALING_3D_MODE_FSR2);
```

Comment:
One of my pet peeves is that we don't do any sanitation on the 3D scaling mode, It is possible for someone to select FSR2 as the scaling mode because they use that on desktop, then also publish their game to mobile using the Vulkan mobile renderer and the scaling mode remains FSR2 but in reality bilinear scaling is used. Also target size currently is not set correctly on the mobile renderer but that is something I am planning on fixing.

Be that as it may, this code will thus break in that situation on the mobile renderer. 

At the moment you can "work around this" by also checking `can_use_storage` which will be false on the mobile renderer.

Reply:
I'm with you, I don't like this handling either. I really think we need to boil down this setting to something like "using upscaling" as well in case other upscalers are added too.

Reply:
I think we need some way to determine which scalers are available. That the mobile renderer currently only supports standard linear scaling doesn't mean we won't support other at some point. GSR was mentioned as a possibility

Reply:
FSR 1/2 in their vanilla state are too demanding for mobile, but a [tweaked version of FSR 1](https://github.com/godotengine/godot-proposals/issues/3786) can run well on high-end mobile devices. FSR 2 has also been used on Switch in No Man's Sky, but it had to be significantly tweaked for this and these tweaks aren't publicly available.

Reply:
I moved the condition to check for the upscaled texture instead and moved the initialization check in the forward clustered renderer to be before the copy effects are performed. This should fix the condition nicely for mobile.